From e408ece863d7a8ef294a8a99fea9d9bd4f175c04 Mon Sep 17 00:00:00 2001
From: liubo <liubo254@huawei.com>
Date: Wed, 2 Mar 2022 18:25:36 +0800
Subject: [PATCH] etmem: etmem scan module Replace WARN_ONCE() with
 debug_printk for "nothing read"
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: e408ece863d7a8ef294a8a99fea9d9bd4f175c04
Modified-by-SEL: No


euleros inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I4VUFS
CVE: NA

-------------------------------------------------
The etmem_scan.ko module is used to scan the process memory.

The specific usage is as follows:
The etmem user mode process issues scan commands through
/proc/pid/idle_pages, and the etmem_scan module scans
based on the issued address information.

Under certain circumstances, the phenomenon that the scan result
is empty may occur. This phenomenon is a normal logic flow and
does not need to print the log through WARN_ONCE.

Therefore, Replace WARN_ONCE() with debug_printk for "nothing read"

Signed-off-by: liubo <liubo254@huawei.com>
Reviewed-by: Miaohe Lin <linmiaohe@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/proc/etmem_scan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/proc/etmem_scan.c b/fs/proc/etmem_scan.c
index ec06e606ca7b..8bcb8d3af7c5 100644
--- a/fs/proc/etmem_scan.c
+++ b/fs/proc/etmem_scan.c
@@ -1244,7 +1244,7 @@ static int mm_idle_walk_range(struct page_idle_ctrl *pic,
 					 pic->next_hva, end);
 		ret = 0;
 	} else
-		WARN_ONCE(1, "nothing read");
+		debug_printk("nothing read");
 	return ret;
 }
 
-- 
2.26.2

